bin/init.ml

```diff
@@ -30,17 +30,14 @@ let atom_printer ppf a = Format.pp_print_string ppf (Dune_lang.Atom.to_string a)
 let component_name_parser s =
```

Comment:
@shonfeder why is this called "component" rather than "library"?

Reply:
iirc, this is because it should parse names for any "component", including executables, projects, and tests. The idea being that dune build many kinds of components -- not just libraries -- and the `init` should (in theory) be extensible to initialize any of them.

Reply:
Ah I see, that makes sense. Thanks! This probably deserve a comment, otherwise the usage of the `Lib_name....` tends to indicate that this is for parsing library names.